### PR TITLE
worker server proxy

### LINKED ISSUE #62: WorkerServerProxy
The proxy which represents the Server on the Worker.
Related to #61

### PR DESCRIPTION
**Pull request by nikolatzotchev**
before I write any tests, is this what the worker proxy is supposed to be? @JohannesGaessler 

Fixes #62

### CODE REVIEW

File: src/ja/worker/proxy/proxy.py [hunk 1]
```diff
@@ -55,14 +61,27 @@ class WorkerServerProxy(IWorkerServerProxy):
...
-        pass
+    def register_self(self, uid: str, work_machine_resources: WorkMachineResources) -> Response:
+        wm: WorkMachine = WorkMachine(uid, WorkMachineState.ONLINE, work_machine_resources)
+        register_command: RegisterWorkerCommand = RegisterWorkerCommand(wm)
+        response = self._ssh_connection.send_server_command(register_command)
+        return response
 
-    def notify_job_finished(self, uid: str) -> ServerResponse:
-        pass
+    # maybe rename to retire self or?
```
Comment (JohannesGaessler) [inline src/ja/worker/proxy/proxy.py]: I don't think the name matters much.
If you want to rename it please wait until Joe has merged his PR.

**Comment by nikolatzotchev** [comment]
One more question is it really useful to write test for that, because I feel I just make dummies and don't really test anything.

File: src/ja/worker/proxy/proxy.py [hunk 2]
```diff
@@ -55,14 +61,27 @@ class WorkerServerProxy(IWorkerServerProxy):
...
-    def register_self(self, work_machine_resources: WorkMachineResources) -> ServerResponse:
+    def _get_ssh_connection(self, ssh_config: SSHConfig) -> ISSHConnection:
         pass
 
-    def unregister_self(self) -> ServerResponse:
-        pass
+    def register_self(self, uid: str, work_machine_resources: WorkMachineResources) -> Response:
+        wm: WorkMachine = WorkMachine(uid, WorkMachineState.ONLINE, work_machine_resources)
+        register_command: RegisterWorkerCommand = RegisterWorkerCommand(wm)
+        response = self._ssh_connection.send_server_command(register_command)
```
Comment (M1keReck) [inline src/ja/worker/proxy/proxy.py]: Where did you instantiate self._ssh_connection?

Comment (nikolatzotchev) [reply src/ja/worker/proxy/proxy.py]: It is initialized in the parent class.

**Pull request merged**